prov/psm/src/psmx_fabric.c

```diff
@@ -103,10 +100,23 @@ int psmx_fabric(struct fi_fabric_attr *attr,
 	if (!fabric_priv)
 		return -FI_ENOMEM;
 
+	psmx_get_uuid(fabric_priv->uuid);
+	if (psmx_env.name_server) {
+		fabric_priv->util_fabric.name_server =
+			calloc(1, sizeof(*fabric_priv->util_fabric.name_server));
+		if (!fabric_priv->util_fabric.name_server) {
+			free(fabric_priv);
+			return -FI_ENOMEM;
+		}
+		fabric_priv->util_fabric.name_server->ns_port =
+			psmx_uuid_to_port(fabric_priv->uuid);
+	}
```

Comment:
It should be separate. Many providers will not need the ns functionality.  Let the provider pass in whatever is needed into start().  I don't know that we even need to store the ns pointer as part of the fabric structure.  Maybe the provider only needs to start one ns, not one per opened fabric.